code/modules/mob/living/carbon/carbon_movement.dm

```diff
@@ -64,11 +64,22 @@
 	. = ..()
 	if(isnull(.))
 		return
-	if(. & !(FLYING | FLOATING))
+	if(!(. & (FLYING | FLOATING)))
 		if(movement_type & (FLYING | FLOATING)) //From not flying to flying.
+			remove_movespeed_modifier(/datum/movespeed_modifier/limbless)
 			REMOVE_TRAIT(src, TRAIT_FLOORED, LACKING_LOCOMOTION_APPENDAGES_TRAIT)
 			REMOVE_TRAIT(src, TRAIT_IMMOBILIZED, LACKING_LOCOMOTION_APPENDAGES_TRAIT)
```

Comment:
To clarify here, these traits aren't used in dealing with movespeed right? I feel like the update should come after, not before them.

Reply:
What do you mean? We'd want to normalize speed before letting the mob move, no?

Reply:
Well those traits aren't checked in the movespeed stuff are they? Cause if they are you've sorta invalidated the event system.